IndentationCheck: problem with content of anonymous class

`/var/tmp $ javac ManageHightPower.java`:

```
No output. Success.
```

`/var/tmp $ cat ManageHighPower.java`:

```
package com.android.tv.settings.device.apps.specialaccess;

public class ManageHighPower {

    private final StringBuffer mFilter =
            new StringBuffer(
                    new CharSequence() {
                        @Override                              // line 8
                        public char charAt(int index) {
                            return 'A';
                        }

                        public int length() {
                            return 1;
                        }

                        public CharSequence subSequence(int start, int end) {
                            return this;
                        }

                        public String toString() {
                            return "Foo";
                        }
                    });
}
```

`/var/tmp $ cat config.xml`:

```
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
      <module name="TreeWalker">
          <module name="Indentation">
              <property name="severity" value="error"/>
              <property name="basicOffset" value="4"/>
              <property name="throwsIndent" value="8"/>
              <property name="lineWrappingIndentation" value="8"/>
          </module>
      </module>
</module>
```

`/var/tmp $ java -Duser.language=en -Duser.country=US -jar checkstyle-8.5-all.jar -c config.xml ManageHighPower.java`:

```
Starting audit...
[ERROR] .../ManageHighPower.java:8: 'method def modifier' has incorrect indentation level 24, expected level should be one of the following: 12, 16, 20. [Indentation]
[ERROR] .../ManageHighPower.java:10: 'method def' child has incorrect indentation level 28, expected level should be one of the following: 16, 20, 24. [Indentation]
[ERROR] .../ManageHighPower.java:11: 'method def rcurly' has incorrect indentation level 24, expected level should be one of the following: 12, 16, 20. [Indentation]
[ERROR] .../ManageHighPower.java:13: 'method def modifier' has incorrect indentation level 24, expected level should be one of the following: 12, 16, 20. [Indentation]
[ERROR] .../ManageHighPower.java:14: 'method def' child has incorrect indentation level 28, expected level should be one of the following: 16, 20, 24. [Indentation]
[ERROR] .../ManageHighPower.java:15: 'method def rcurly' has incorrect indentation level 24, expected level should be one of the following: 12, 16, 20. [Indentation]
[ERROR] .../ManageHighPower.java:17: 'method def modifier' has incorrect indentation level 24, expected level should be one of the following: 12, 16, 20. [Indentation]
[ERROR] .../ManageHighPower.java:18: 'method def' child has incorrect indentation level 28, expected level should be one of the following: 16, 20, 24. [Indentation]
[ERROR] .../ManageHighPower.java:19: 'method def rcurly' has incorrect indentation level 24, expected level should be one of the following: 12, 16, 20. [Indentation]
[ERROR] .../ManageHighPower.java:21: 'method def modifier' has incorrect indentation level 24, expected level should be one of the following: 12, 16, 20. [Indentation]
[ERROR] .../ManageHighPower.java:22: 'method def' child has incorrect indentation level 28, expected level should be one of the following: 16, 20, 24. [Indentation]
[ERROR] .../Desktop/ManageHighPower.java:23: 'method def rcurly' has incorrect indentation level 24, expected level should be one of the following: 12, 16, 20. [Indentation]
[ERROR] .../ManageHighPower.java:24: 'object def rcurly' has incorrect indentation level 20, expected level should be one of the following: 8, 12, 16. [Indentation]
Audit done.
Checkstyle ends with 13 errors.
```

---------------

I expected the indention to be correct without any warnings.

--------------

--- a/src/it/java/com/google/checkstyle/test/chapter4formatting/rule4841indentation/IndentationTest.java
+++ b/src/it/java/com/google/checkstyle/test/chapter4formatting/rule4841indentation/IndentationTest.java
@@ -81,6 +81,17 @@ public class IndentationTest extends AbstractIndentationTestSupport {
     }
 
     @Test
+    public void testCorrectNewKeyword() throws Exception {
+        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+
+        final Configuration checkConfig = getModuleConfig("Indentation");
+        final String filePath = getPath("InputIndentationCorrectNewChildren.java");
+
+        final Integer[] warnList = getLinesWithWarn(filePath);
+        verify(checkConfig, filePath, expected, warnList);
+    }
+
+    @Test
     public void testCorrect() throws Exception {
         final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
 

--- a/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/InputIndentationCorrectIfAndParameter.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule42blockindentation/InputIndentationCorrectIfAndParameter.java
@@ -16,14 +16,14 @@ class InputIndentationCorrectIfAndParameter { //indent:0 exp:0
     if (conditionSecond(10000000000.0, new //indent:4 exp:4
         SecondClassLongNam3("Looooooooooooo" //indent:8 exp:8
         + "oooooooooooong").getString(new FooIfClass(), //indent:8 exp:8
-        new SecondClassLongNam3("loooooooooong"). //indent:8 exp:8
+        new SecondClassLongNam3("loooooooooong"). //indent:8 exp:10,12 warn
         getInteger(new FooIfClass(), "loooooooooooooong")), "loooooooooooong") //indent:8 exp:8
         || conditionThird(2048) || conditionFourth(new //indent:8 exp:8
         SecondClassLongNam3("Looooooooooooooo" //indent:8 exp:8
         + "ooooooooooooong").getBoolean(new FooIfClass(), false)) || //indent:8 exp:8
         conditionFifth(true, new SecondClassLongNam3(getString(2048, "Looo" //indent:8 exp:8
         + "ooooooooooooooooooooooooooooooooooooooooooong")).getBoolean( //indent:8 exp:8
-        new FooIfClass(), true)) || conditionSixth(false, new //indent:8 exp:8
+        new FooIfClass(), true)) || conditionSixth(false, new //indent:8 exp:10,12 warn
         SecondClassLongNam3(getString(100000, "Loooooong" //indent:8 exp:8
         + "Fooooooo><"))) || conditionNoArg() //indent:8 exp:8
         || conditionNoArg() || //indent:8 exp:8
@@ -72,14 +72,14 @@ class InputIndentationCorrectIfAndParameter { //indent:0 exp:0
       if (conditionSecond(10000000000.0, new //indent:6 exp:6
           SecondClassLongNam3("Looooooooooooo" //indent:10 exp:10
           + "oooooooooooong").getString(new FooIfClass(), //indent:10 exp:10
-          new SecondClassLongNam3("loooooooooong"). //indent:10 exp:10
+          new SecondClassLongNam3("loooooooooong"). //indent:10 exp:12,14 warn
           getInteger(new FooIfClass(), "loooooooooooooong")), "loooooooooooong") //indent:10 exp:10
           || conditionThird(2048) || conditionFourth(new //indent:10 exp:10
           SecondClassLongNam3("Looooooooooooooo" //indent:10 exp:10
             + "ooooooooooooong").getBoolean(new FooIfClass(), false)) || //indent:12 exp:>=10
             conditionFifth(true, new SecondClassLongNam3(getString(2048, "Looo" //indent:12 exp:>=10
             + "ooooooooooooooooooooooooooooooooooooooooooong")).getBoolean( //indent:12 exp:>=10
-            new FooIfClass(), true)) || conditionSixth(false, new //indent:12 exp:>=10
+            new FooIfClass(), true)) || conditionSixth(false, new //indent:12 exp:14,16 warn
               SecondClassLongNam3(getString(100000, "Loooooong" //indent:14 exp:>=10
               + "Fooooooo><"))) || conditionNoArg() //indent:14 exp:>=10
               || conditionNoArg() || //indent:14 exp:>=10
@@ -100,14 +100,14 @@ class InputIndentationCorrectIfAndParameter { //indent:0 exp:0
         if (conditionSecond(10000000000.0, new //indent:8 exp:8
             SecondClassLongNam3("Looooooooooooo" //indent:12 exp:12
             + "oooooooooooong").getString(new FooIfClass(), //indent:12 exp:12
-            new SecondClassLongNam3("loooooooooong"). //indent:12 exp:12
+            new SecondClassLongNam3("loooooooooong"). //indent:12 exp:14,16 warn
             getInteger(new FooIfClass(), "looooooooooooong")), "loooooooooooong") //indent:12 exp:12
             || conditionThird(2048) || conditionFourth(new //indent:12 exp:12
             SecondClassLongNam3("Looooooooooooooo" //indent:12 exp:12
               + "ooooooooooooong").getBoolean(new FooIfClass(), false)) || //indent:14 exp:>=12
               conditionFifth(true, new SecondClassLongNam3(getString(2048, "Lo" //indent:14 exp:>=12
               + "ooooooooooooooooooooooooooooooooooooooooooong")).getBoolean( //indent:14 exp:>=12
-              new FooIfClass(), true)) || conditionSixth(false, new //indent:14 exp:>=12
+              new FooIfClass(), true)) || conditionSixth(false, new //indent:14 exp:16,18 warn
                 SecondClassLongNam3(getString(100000, "Loooooong" //indent:16 exp:>=12
                 + "Fooooooo><"))) || conditionNoArg() //indent:16 exp:>=12
                 || conditionNoArg() || //indent:16 exp:>=12

--- a/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4841indentation/InputIndentationCorrect.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4841indentation/InputIndentationCorrect.java
@@ -36,27 +36,28 @@ public abstract class InputIndentationCorrect { //indent:0 exp:0
     /** Matches no characters. */ //indent:4 exp:4
     public static final InputFastMatcher NONE = //indent:4 exp:4
             new InputFastMatcher() { //indent:12 exp:>=8
-      @Override public boolean matches(char c) { //indent:6 exp:6
-        return false; //indent:8 exp:8
-      } //indent:6 exp:6
+              @Override public boolean matches(char c) { //indent:14 exp:14
+                return false; //indent:16 exp:16
+              } //indent:14 exp:14
 
-      @Override public String replaceFrom(CharSequence seq, CharSequence repl) { //indent:6 exp:6
-        checkNotNull(repl); //indent:8 exp:8
-        return seq.toString(); //indent:8 exp:8
-      } //indent:6 exp:6
+              @Override public String replaceFrom(CharSequence seq, //indent:14 exp:14
+                                                  CharSequence repl) { //indent:50 exp:50
+                checkNotNull(repl); //indent:16 exp:16
+                return seq.toString(); //indent:16 exp:16
+              } //indent:14 exp:14
 
-      private void checkNotNull(CharSequence replacement) {} //indent:6 exp:6
+              private void checkNotNull(CharSequence replacement) {} //indent:14 exp:14
 
-      @Override public String collapseFrom(CharSequence sequence, //indent:6 exp:6
-          char replacement) { //indent:10 exp:10
-        return sequence.toString(); //indent:8 exp:8
-      } //indent:6 exp:6
+              @Override public String collapseFrom(CharSequence sequence, //indent:14 exp:14
+                  char replacement) { //indent:18 exp:18
+                return sequence.toString(); //indent:16 exp:16
+              } //indent:14 exp:14
 
-      @Override //indent:6 exp:6
-      public String trimTrailingFrom(CharSequence sequence) { //indent:6 exp:6
-        return sequence.toString(); //indent:8 exp:8
-      } //indent:6 exp:6
-    }; //indent:4 exp:4
+              @Override //indent:14 exp:14
+              public String trimTrailingFrom(CharSequence sequence) { //indent:14 exp:14
+                return sequence.toString(); //indent:16 exp:16
+              } //indent:14 exp:14
+            }; //indent:12 exp:12
   } //indent:2 exp:2
 
   public boolean matches(char c) { //indent:2 exp:2

--- a/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4841indentation/InputIndentationCorrectClass.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4841indentation/InputIndentationCorrectClass.java
@@ -28,7 +28,7 @@ public class InputIndentationCorrectClass  //indent:0 exp:0
     SecondClassWithLongLongLongLongName anon =  //indent:4 exp:4
           new SecondClassWithLongLongLongLongName() {  //indent:10 exp:>=8
 
-    };  //indent:4 exp:4
+          };  //indent:10 exp:10
 
     SecondClassWithLongLongLongLongName anon2 = new  //indent:4 exp:4
           SecondClassWithLongLongLongLongName() {  //indent:10 exp:>=8

--- a/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4841indentation/InputIndentationCorrectFieldAndParameter.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4841indentation/InputIndentationCorrectFieldAndParameter.java
@@ -10,15 +10,15 @@ class InputIndentationCorrectFieldAndParameter { //indent:0 exp:0
   boolean secondFlag = conditionSecond(10000000000.0, new //indent:2 exp:2
       SecondFieldLongName("Looooooooooooo" //indent:6 exp:6
       + "oooooooooooong").getString(new FooFieldClass(), //indent:6 exp:6
-      new SecondFieldLongName("loooooooooong"). //indent:6 exp:6
+          new SecondFieldLongName("loooooooooong"). //indent:10 exp:10
       getInteger(new FooFieldClass(), "loooooooooooooong")), "loooooooooooong") //indent:6 exp:6
       || conditionThird(2048) || conditionFourth(new //indent:6 exp:6
       SecondFieldLongName("Looooooooooooooo" //indent:6 exp:6
       + "ooooooooooooong").getBoolean(new FooFieldClass(), false)) || //indent:6 exp:6
       conditionFifth(true, new SecondFieldLongName(getString(2048, "Looo" //indent:6 exp:6
       + "ooooooooooooooooooooooooooooooooooooooooooong")).getBoolean( //indent:6 exp:6
-      new FooFieldClass(), true)) || conditionSixth(false, new //indent:6 exp:6
-      SecondFieldLongName(getString(100000, "Loooooong" //indent:6 exp:6
+          new FooFieldClass(), true)) || conditionSixth(false, new //indent:10 exp:10
+          SecondFieldLongName(getString(100000, "Loooooong" //indent:10 exp:10
       + "Fooooooo><"))) || conditionNoArg() //indent:6 exp:6
       || conditionNoArg() || //indent:6 exp:6
       conditionNoArg() || conditionNoArg(); //indent:6 exp:6
@@ -68,15 +68,15 @@ class InputIndentationCorrectFieldAndParameter { //indent:0 exp:0
     boolean secondFlag = conditionSecond(10000000000.0, new //indent:4 exp:4
         SecondFieldLongName("Looooooooooooo" //indent:8 exp:8
            + "oooooooooooong").getString(new FooFieldClass(), //indent:11 exp:>=8
-          new SecondFieldLongName("loooooooooong"). //indent:10 exp:>=8
+             new SecondFieldLongName("loooooooooong"). //indent:13 exp:13
             getInteger(new FooFieldClass(), "looooooooong")), "loooooooooooong") //indent:12 exp:>=8
              || conditionThird(2048) || conditionFourth(new //indent:13 exp:>=8
                 SecondFieldLongName("Looooooooooooooo" //indent:16 exp:>=8
             + "ooooooooooooong").getBoolean(new FooFieldClass(), false)) || //indent:12 exp:>=8
            conditionFifth(true, new SecondFieldLongName(getString(2048, "Looo" //indent:11 exp:>=8
                + "ooooooooooooooooooooooooooooooooooooooooooong")).getBoolean( //indent:15 exp:>=8
-            new FooFieldClass(), true)) || conditionSixth(false, new //indent:12 exp:>=8
-            SecondFieldLongName(getString(100000, "Loooooong" //indent:12 exp:>=8
+                 new FooFieldClass(), true)) || conditionSixth(false, new //indent:17 exp:>=17
+                 SecondFieldLongName(getString(100000, "Loooooong" //indent:17 exp:17
                 + "Fooooooo><"))) || conditionNoArg() //indent:16 exp:>=8
           || conditionNoArg() || //indent:10 exp:>=8
           conditionNoArg() || conditionNoArg(); //indent:10 exp:>=8
@@ -92,8 +92,8 @@ class InputIndentationCorrectFieldAndParameter { //indent:0 exp:0
              + "ooooooooooooong").getBoolean(new FooFieldClass(), false)) || //indent:13 exp:>=10
             conditionFifth(true, new SecondFieldLongName(getString(2048, "Looo" //indent:12 exp:>=10
                 + "ooooooooooooooooooooooooooooooooooooooooooong")).getBoolean( //indent:16 exp:>=10
-             new FooFieldClass(), true)) || conditionSixth(false, new //indent:13 exp:>=8
-             SecondFieldLongName(getString(100000, "Loooooong" //indent:13 exp:>=10
+                  new FooFieldClass(), true)) || conditionSixth(false, new //indent:18 exp:>=18
+                  SecondFieldLongName(getString(100000, "Loooooong" //indent:18 exp:>=18
                  + "Fooooooo><"))) || conditionNoArg() //indent:17 exp:>=10
            || conditionNoArg() || //indent:11 exp:>=10
            conditionNoArg() || conditionNoArg(); //indent:11 exp:>=10

--- a/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4841indentation/InputIndentationCorrectIfAndParameter.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4841indentation/InputIndentationCorrectIfAndParameter.java
@@ -16,15 +16,15 @@ class InputIndentationCorrectIfAndParameter { //indent:0 exp:0
     if (conditionSecond(10000000000.0, new //indent:4 exp:4
         SecondClassLongNam6("Looooooooooooo" //indent:8 exp:8
         + "oooooooooooong").getString(new FooIfClass(), //indent:8 exp:8
-        new SecondClassLongNam6("loooooooooong"). //indent:8 exp:8
+          new SecondClassLongNam6("loooooooooong"). //indent:10 exp:10
         getInteger(new FooIfClass(), "loooooooooooooong")), "loooooooooooong") //indent:8 exp:8
         || conditionThird(2048) || conditionFourth(new //indent:8 exp:8
         SecondClassLongNam6("Looooooooooooooo" //indent:8 exp:8
         + "ooooooooooooong").getBoolean(new FooIfClass(), false)) || //indent:8 exp:8
         conditionFifth(true, new SecondClassLongNam6(getString(2048, "Looo" //indent:8 exp:8
         + "ooooooooooooooooooooooooooooooooooooooooooong")).getBoolean( //indent:8 exp:8
-        new FooIfClass(), true)) || conditionSixth(false, new //indent:8 exp:8
-        SecondClassLongNam6(getString(100000, "Loooooong" //indent:8 exp:8
+          new FooIfClass(), true)) || conditionSixth(false, new //indent:10 exp:10
+          SecondClassLongNam6(getString(100000, "Loooooong" //indent:10 exp:10
         + "Fooooooo><"))) || conditionNoArg() //indent:8 exp:8
         || conditionNoArg() || //indent:8 exp:8
         conditionNoArg() || conditionNoArg()) {} //indent:8 exp:8
@@ -72,14 +72,14 @@ class InputIndentationCorrectIfAndParameter { //indent:0 exp:0
       if (conditionSecond(10000000000.0, new //indent:6 exp:6
           SecondClassLongNam6("Looooooooooooo" //indent:10 exp:10
           + "oooooooooooong").getString(new FooIfClass(), //indent:10 exp:10
-          new SecondClassLongNam6("loooooooooong"). //indent:10 exp:10
+            new SecondClassLongNam6("loooooooooong"). //indent:12 exp:12
           getInteger(new FooIfClass(), "loooooooooooooong")), "loooooooooooong") //indent:10 exp:10
           || conditionThird(2048) || conditionFourth(new //indent:10 exp:10
           SecondClassLongNam6("Looooooooooooooo" //indent:10 exp:10
             + "ooooooooooooong").getBoolean(new FooIfClass(), false)) || //indent:12 exp:>=10
             conditionFifth(true, new SecondClassLongNam6(getString(2048, "Looo" //indent:12 exp:>=10
             + "ooooooooooooooooooooooooooooooooooooooooooong")).getBoolean( //indent:12 exp:>=10
-            new FooIfClass(), true)) || conditionSixth(false, new //indent:12 exp:>=10
+              new FooIfClass(), true)) || conditionSixth(false, new //indent:14 exp:14,16
               SecondClassLongNam6(getString(100000, "Loooooong" //indent:14 exp:>=10
               + "Fooooooo><"))) || conditionNoArg() //indent:14 exp:>=10
               || conditionNoArg() || //indent:14 exp:>=10
@@ -100,14 +100,14 @@ class InputIndentationCorrectIfAndParameter { //indent:0 exp:0
         if (conditionSecond(10000000000.0, new //indent:8 exp:8
             SecondClassLongNam6("Looooooooooooo" //indent:12 exp:12
             + "oooooooooooong").getString(new FooIfClass(), //indent:12 exp:12
-            new SecondClassLongNam6("loooooooooong"). //indent:12 exp:12
+              new SecondClassLongNam6("loooooooooong"). //indent:14 exp:14
             getInteger(new FooIfClass(), "looooooooooooong")), "loooooooooooong") //indent:12 exp:12
             || conditionThird(2048) || conditionFourth(new //indent:12 exp:12
             SecondClassLongNam6("Looooooooooooooo" //indent:12 exp:12
               + "ooooooooooooong").getBoolean(new FooIfClass(), false)) || //indent:14 exp:>=12
               conditionFifth(true, new SecondClassLongNam6(getString(2048, "Lo" //indent:14 exp:>=12
               + "ooooooooooooooooooooooooooooooooooooooooooong")).getBoolean( //indent:14 exp:>=12
-              new FooIfClass(), true)) || conditionSixth(false, new //indent:14 exp:>=12
+                new FooIfClass(), true)) || conditionSixth(false, new //indent:16 exp:>=16
                 SecondClassLongNam6(getString(100000, "Loooooong" //indent:16 exp:>=12
                 + "Fooooooo><"))) || conditionNoArg() //indent:16 exp:>=12
                 || conditionNoArg() || //indent:16 exp:>=12

--- a/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4841indentation/InputIndentationCorrectNewChildren.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4841indentation/InputIndentationCorrectNewChildren.java
@@ -1,0 +1,25 @@
+package com.google.checkstyle.test.chapter4formatting.rule4841indentation; //indent:0 exp:0
+
+public class InputIndentationCorrectNewChildren { //indent:0 exp:0
+
+  private final StringBuffer mFilter = //indent:2 exp:2
+      new StringBuffer( //indent:6 exp:6
+          new CharSequence() { //indent:10 exp:10
+            @Override //indent:12 exp:12
+            public char charAt(int index) { //indent:12 exp:12
+              return 'A'; //indent:14 exp:14
+            } //indent:12 exp:12
+
+            public int length() { //indent:12 exp:12
+              return 1; //indent:14 exp:14
+            } //indent:12 exp:12
+
+            public CharSequence subSequence(int start, int end) { //indent:12 exp:12
+              return this; //indent:14 exp:14
+            } //indent:12 exp:12
+
+            public String toString() { //indent:12 exp:12
+              return "Foo"; //indent:14 exp:14
+            } //indent:12 exp:12
+          }); //indent:10 exp:10
+} //indent:0 exp:0

--- a/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4841indentation/InputIndentationCorrectReturnAndParameter.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4841indentation/InputIndentationCorrectReturnAndParameter.java
@@ -11,15 +11,15 @@ class InputIndentationCorrectReturnAndParameter { //indent:0 exp:0
     return conditionSecond(10000000000.0, new //indent:4 exp:4
         SecondClassLongName("Looooooooooooo" //indent:8 exp:8
         + "oooooooooooong").getString(new FooReturnClass(), //indent:8 exp:8
-        new SecondClassLongName("loooooooooong"). //indent:8 exp:8
+          new SecondClassLongName("loooooooooong"). //indent:10 exp:10
         getInteger(new FooReturnClass(), "loooooooooooooong")), "loooooooooooong") //indent:8 exp:8
         || conditionThird(2048) || conditionFourth(new //indent:8 exp:8
         SecondClassLongName("Looooooooooooooo" //indent:8 exp:8
         + "ooooooooooooong").getBoolean(new FooReturnClass(), false)) || //indent:8 exp:8
         conditionFifth(true, new SecondClassLongName(getString(2048, "Looo" //indent:8 exp:8
         + "ooooooooooooooooooooooooooooooooooooooooooong")).getBoolean( //indent:8 exp:8
-        new FooReturnClass(), true)) || conditionSixth(false, new //indent:8 exp:8
-        SecondClassLongName(getString(100000, "Loooooong" //indent:8 exp:8
+          new FooReturnClass(), true)) || conditionSixth(false, new //indent:10 exp:10
+          SecondClassLongName(getString(100000, "Loooooong" //indent:10 exp:10
         + "Fooooooo><"))) || conditionNoArg() //indent:8 exp:8
         || conditionNoArg() || //indent:8 exp:8
         conditionNoArg() || conditionNoArg();//indent:8 exp:8
@@ -69,14 +69,14 @@ class InputIndentationCorrectReturnAndParameter { //indent:0 exp:0
       return conditionSecond(10000000000.0, new //indent:6 exp:6
           SecondClassLongName("Looooooooooooo" //indent:10 exp:10
           + "oooooooooooong").getString(new FooReturnClass(), //indent:10 exp:10
-          new SecondClassLongName("loooooooooong"). //indent:10 exp:10
+            new SecondClassLongName("loooooooooong"). //indent:12 exp:12
           getInteger(new FooReturnClass(), "looooooooooong")), "loooooooooooong") //indent:10 exp:10
           || conditionThird(2048) || conditionFourth(new //indent:10 exp:10
           SecondClassLongName("Looooooooooooooo" //indent:10 exp:10
             + "ooooooooooooong").getBoolean(new FooReturnClass(), false)) || //indent:12 exp:>=10
             conditionFifth(true, new SecondClassLongName(getString(2048, "Looo" //indent:12 exp:>=10
             + "ooooooooooooooooooooooooooooooooooooooooooong")).getBoolean( //indent:12 exp:>=10
-            new FooReturnClass(), true)) || conditionSixth(false, new //indent:12 exp:>=10
+              new FooReturnClass(), true)) || conditionSixth(false, new //indent:14 exp:>=14
               SecondClassLongName(getString(100000, "Loooooong" //indent:14 exp:>=10
               + "Fooooooo><"))) || conditionNoArg() //indent:14 exp:>=10
               || conditionNoArg() || //indent:14 exp:>=10
@@ -92,14 +92,14 @@ class InputIndentationCorrectReturnAndParameter { //indent:0 exp:0
         return conditionSecond(10000000000.0, new //indent:8 exp:8
             SecondClassLongName("Looooooooooooo" //indent:12 exp:12
             + "oooooooooooong").getString(new FooReturnClass(), //indent:12 exp:12
-            new SecondClassLongName("loooooooooong"). //indent:12 exp:12
+              new SecondClassLongName("loooooooooong"). //indent:14 exp:14
             getInteger(new FooReturnClass(), "looooooooong")), "loooooooooooong") //indent:12 exp:12
             || conditionThird(2048) || conditionFourth(new //indent:12 exp:12
             SecondClassLongName("Looooooooooooooo" //indent:12 exp:12
               + "ooooooooooooong").getBoolean(new FooReturnClass(), false)) || //indent:14 exp:>=12
               conditionFifth(true, new SecondClassLongName(getString(2048, "Lo" //indent:14 exp:>=12
               + "ooooooooooooooooooooooooooooooooooooooooooong")).getBoolean( //indent:14 exp:>=12
-              new FooReturnClass(), true)) || conditionSixth(false, new //indent:14 exp:>=12
+                new FooReturnClass(), true)) || conditionSixth(false, new //indent:16 exp:>=16
                 SecondClassLongName(getString(100000, "Loooooong" //indent:16 exp:>=12
                 + "Fooooooo><"))) || conditionNoArg() //indent:16 exp:>=12
                 || conditionNoArg() || //indent:16 exp:>=12

--- a/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4841indentation/InputIndentationCorrectWhileDoWhileAndParameter.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4841indentation/InputIndentationCorrectWhileDoWhileAndParameter.java
@@ -31,8 +31,8 @@ class InputIndentationCorrectWhileDoWhileAndParameter { //indent:0 exp:0
             conditionFifth(true, new SecondWhileLongName( //indent:12 exp:>=8
             getString(2048, "Looo" //indent:12 exp:>=8
         + "ooooooooooooooooooooooooooooooooooooooooooong")).getBoolean( //indent:8 exp:8
-        new FooWhileClass(), true)) || conditionSixth(false, new //indent:8 exp:8
-           SecondWhileLongName(getString(100000, "Loooooong" //indent:11 exp:>=8
+          new FooWhileClass(), true)) || conditionSixth(false, new //indent:10 exp:10
+           SecondWhileLongName(getString(100000, "Loooooong" //indent:11 exp:>=10
         + "Fooooooo><"))) || conditionNoArg() //indent:8 exp:8
           || conditionNoArg() || //indent:10 exp:>=8
           conditionNoArg() || conditionNoArg()) {} //indent:10 exp:10
@@ -42,15 +42,15 @@ class InputIndentationCorrectWhileDoWhileAndParameter { //indent:0 exp:0
     } while (conditionSecond(10000000000.0, new //indent:4 exp:4
           SecondWhileLongName("Looooooooooooo" //indent:10 exp:>=8
           + "oooooooooooong").getString(new FooWhileClass(), //indent:10 exp:>=8
-        new SecondWhileLongName("loooooooooong"). //indent:8 exp:8
+            new SecondWhileLongName("loooooooooong"). //indent:12 exp:12
              getInteger(new FooWhileClass(), "loooooooong")), "loooooooooooong") //indent:13 exp:>=8
         || conditionThird(2048) || conditionFourth(new //indent:8 exp:8
            SecondWhileLongName("Looooooooooooooo" //indent:11 exp:>=8
         + "ooooooooooooong").getBoolean(new FooWhileClass(), false)) || //indent:8 exp:8
         conditionFifth(true, new SecondWhileLongName(getString(2048, "Looo" //indent:8 exp:8
         + "ooooooooooooooooooooooooooooooooooooooooooong")).getBoolean( //indent:8 exp:8
-        new FooWhileClass(), true)) || conditionSixth(false, new //indent:8 exp:8
-           SecondWhileLongName(getString(100000, "Loooooong" //indent:11 exp:>=8
+            new FooWhileClass(), true)) || conditionSixth(false, new //indent:12 exp:12
+            SecondWhileLongName(getString(100000, "Loooooong" //indent:12 exp:12
         + "Fooooooo><"))) || conditionNoArg() //indent:8 exp:8
              || conditionNoArg() || //indent:13 exp:>=8
         conditionNoArg() || conditionNoArg()); //indent:8 exp:8
@@ -105,14 +105,14 @@ class InputIndentationCorrectWhileDoWhileAndParameter { //indent:0 exp:0
       while (conditionSecond(10000000000.0, new //indent:6 exp:6
           SecondWhileLongName("Looooooooooooo" //indent:10 exp:10
           + "oooooooooooong").getString(new FooWhileClass(), //indent:10 exp:10
-          new SecondWhileLongName("loooooooooong"). //indent:10 exp:10
+            new SecondWhileLongName("loooooooooong"). //indent:12 exp:12
           getInteger(new FooWhileClass(), "loooooooooooong")), "loooooooooooong") //indent:10 exp:10
           || conditionThird(2048) || conditionFourth(new //indent:10 exp:10
           SecondWhileLongName("Looooooooooooooo" //indent:10 exp:10
             + "ooooooooooooong").getBoolean(new FooWhileClass(), false)) || //indent:12 exp:>=10
             conditionFifth(true, new SecondWhileLongName(getString(2048, "Looo" //indent:12 exp:>=10
             + "ooooooooooooooooooooooooooooooooooooooooooong")).getBoolean( //indent:12 exp:>=10
-            new FooWhileClass(), true)) || conditionSixth(false, new //indent:12 exp:>=10
+              new FooWhileClass(), true)) || conditionSixth(false, new //indent:14 exp:>=14
               SecondWhileLongName(getString(100000, "Loooooong" //indent:14 exp:>=10
               + "Fooooooo><"))) || conditionNoArg() //indent:14 exp:>=10
               || conditionNoArg() || //indent:14 exp:>=10
@@ -123,14 +123,14 @@ class InputIndentationCorrectWhileDoWhileAndParameter { //indent:0 exp:0
       } while (conditionSecond(10000000000.0, new //indent:6 exp:6
           SecondWhileLongName("Looooooooooooo" //indent:10 exp:10
           + "oooooooooooong").getString(new FooWhileClass(), //indent:10 exp:10
-          new SecondWhileLongName("loooooooooong"). //indent:10 exp:10
+            new SecondWhileLongName("loooooooooong"). //indent:12 exp:12,14
           getInteger(new FooWhileClass(), "loooooooooooong")), "loooooooooooong") //indent:10 exp:10
           || conditionThird(2048) || conditionFourth(new //indent:10 exp:10
           SecondWhileLongName("Looooooooooooooo" //indent:10 exp:10
             + "ooooooooooooong").getBoolean(new FooWhileClass(), false)) || //indent:12 exp:>=10
             conditionFifth(true, new SecondWhileLongName(getString(2048, "Looo" //indent:12 exp:>=10
             + "ooooooooooooooooooooooooooooooooooooooooooong")).getBoolean( //indent:12 exp:>=10
-            new FooWhileClass(), true)) || conditionSixth(false, new //indent:12 exp:>=10
+              new FooWhileClass(), true)) || conditionSixth(false, new //indent:14 exp:>=14
               SecondWhileLongName(getString(100000, "Loooooong" //indent:14 exp:>=10
               + "Fooooooo><"))) || conditionNoArg() //indent:14 exp:>=10
               || conditionNoArg() || //indent:14 exp:>=10
@@ -158,14 +158,14 @@ class InputIndentationCorrectWhileDoWhileAndParameter { //indent:0 exp:0
         while (conditionSecond(10000000000.0, new //indent:8 exp:8
             SecondWhileLongName("Looooooooooooo" //indent:12 exp:12
             + "oooooooooooong").getString(new FooWhileClass(), //indent:12 exp:12
-            new SecondWhileLongName("loooooooooong"). //indent:12 exp:12
+              new SecondWhileLongName("loooooooooong"). //indent:14 exp:14
             getInteger(new FooWhileClass(), "loooooooooong")), "loooooooooooong") //indent:12 exp:12
             || conditionThird(2048) || conditionFourth(new //indent:12 exp:12
             SecondWhileLongName("Looooooooooooooo" //indent:12 exp:12
               + "ooooooooooooong").getBoolean(new FooWhileClass(), false)) || //indent:14 exp:>=12
               conditionFifth(true, new SecondWhileLongName(getString(2048, "Lo" //indent:14 exp:>=12
               + "ooooooooooooooooooooooooooooooooooooooooooong")).getBoolean( //indent:14 exp:>=12
-              new FooWhileClass(), true)) || conditionSixth(false, new //indent:14 exp:>=12
+                new FooWhileClass(), true)) || conditionSixth(false, new //indent:16 exp:16,18
                 SecondWhileLongName(getString(100000, "Loooooong" //indent:16 exp:>=12
                 + "Fooooooo><"))) || conditionNoArg() //indent:16 exp:>=12
                 || conditionNoArg() || //indent:16 exp:>=12
@@ -177,14 +177,14 @@ class InputIndentationCorrectWhileDoWhileAndParameter { //indent:0 exp:0
         } while (conditionSecond(10000000000.0, new //indent:8 exp:8
             SecondWhileLongName("Looooooooooooo" //indent:12 exp:12
             + "oooooooooooong").getString(new FooWhileClass(), //indent:12 exp:12
-            new SecondWhileLongName("loooooooooong"). //indent:12 exp:12
+              new SecondWhileLongName("loooooooooong"). //indent:14 exp:14
             getInteger(new FooWhileClass(), "loooooooooong")), "loooooooooooong") //indent:12 exp:12
             || conditionThird(2048) || conditionFourth(new //indent:12 exp:12
             SecondWhileLongName("Looooooooooooooo" //indent:12 exp:12
               + "ooooooooooooong").getBoolean(new FooWhileClass(), false)) || //indent:14 exp:>=12
               conditionFifth(true, new SecondWhileLongName(getString(2048, "Lo" //indent:14 exp:>=12
               + "ooooooooooooooooooooooooooooooooooooooooooong")).getBoolean( //indent:14 exp:>=12
-              new FooWhileClass(), true)) || conditionSixth(false, new //indent:14 exp:>=12
+                new FooWhileClass(), true)) || conditionSixth(false, new //indent:16 exp:>=16
                 SecondWhileLongName(getString(100000, "Loooooong" //indent:16 exp:>=12
                 + "Fooooooo><"))) || conditionNoArg() //indent:16 exp:>=12
                 || conditionNoArg() || //indent:16 exp:>=12

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/NewHandler.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/NewHandler.java
@@ -21,11 +21,15 @@ package com.puppycrawl.tools.checkstyle.checks.indentation;
 
 import com.puppycrawl.tools.checkstyle.api.DetailAST;
 import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.TokenUtil;
 
 /**
  * Handler for operator new.
  */
 public class NewHandler extends AbstractExpressionHandler {
+
+    /** The AST which is handled by this handler. */
+    private final DetailAST mainAst;
 
     /**
      * Construct an instance of this handler with the given indentation check,
@@ -39,31 +43,79 @@ public class NewHandler extends AbstractExpressionHandler {
                       DetailAST ast,
                       AbstractExpressionHandler parent) {
         super(indentCheck, "operator new", ast, parent);
+        mainAst = ast;
     }
 
     @Override
     public void checkIndentation() {
-        final DetailAST type = getMainAst().getFirstChild();
-        if (type != null) {
-            checkExpressionSubtree(type, getIndent(), false, false);
+        // if new is on the line start and it is not the part of assignment.
+        if (isOnStartOfLine(mainAst)
+                && !isNewKeywordFollowedByAssign()) {
+            final int columnNo = expandedTabsColumnNo(mainAst);
+            final IndentLevel level = getIndentImpl();
+
+            if (columnNo < level.getFirstIndentLevel()) {
+                logError(mainAst, "", columnNo, level);
+            }
         }
 
-        final DetailAST lparen = getMainAst().findFirstToken(TokenTypes.LPAREN);
+        final DetailAST firstChild = mainAst.getFirstChild();
+        if (firstChild != null) {
+            checkExpressionSubtree(firstChild, getIndent(), false, false);
+        }
+
+        final DetailAST lparen = mainAst.findFirstToken(TokenTypes.LPAREN);
         checkLeftParen(lparen);
     }
 
     @Override
-    protected IndentLevel getIndentImpl() {
-        final IndentLevel result;
-        // if our expression isn't first on the line, just use the start
-        // of the line
-        if (getLineStart(getMainAst()) == getMainAst().getColumnNo()) {
-            result = super.getIndentImpl();
+    public IndentLevel getSuggestedChildIndent(AbstractExpressionHandler child) {
+        final int offset;
+        if (TokenUtil.isOfType(child.getMainAst(), TokenTypes.OBJBLOCK)) {
+            offset = getBasicOffset();
         }
         else {
-            result = new IndentLevel(getLineStart(getMainAst()));
+            offset = getLineWrappingIndent();
         }
+        return new IndentLevel(getIndent(), offset);
+    }
+
+    @Override
+    protected IndentLevel getIndentImpl() {
+        IndentLevel result;
+        // if our expression isn't first on the line, just use the start
+        // of the line
+        if (getLineStart(mainAst) == mainAst.getColumnNo()) {
+            result = super.getIndentImpl();
+
+            if (isNewKeywordFollowedByAssign()) {
+                result = new IndentLevel(result, getLineWrappingIndent());
+            }
+        }
+        else {
+            result = new IndentLevel(getLineStart(mainAst));
+        }
+
         return result;
+    }
+
+    /**
+     * Checks if the 'new' keyword is followed by an assignment.
+     *
+     * @return true if new keyword is followed by assignment.
+     */
+    private boolean isNewKeywordFollowedByAssign() {
+        return mainAst.getParent().getParent().getType() == TokenTypes.ASSIGN;
+    }
+
+    /**
+     * A shortcut for {@code IndentationCheck} property.
+     *
+     * @return value of lineWrappingIndentation property
+     *         of {@code IndentationCheck}
+     */
+    private int getLineWrappingIndent() {
+        return getIndentCheck().getLineWrappingIndentation();
     }
 
     @Override

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlLoaderTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlLoaderTest.java
@@ -50,7 +50,7 @@ public class ImportControlLoaderTest {
     public void testLoad() throws CheckstyleException {
         final AbstractImportControl root =
                 ImportControlLoader.load(
-                new File(getPath("InputImportControlLoaderComplete.xml")).toURI());
+                    new File(getPath("InputImportControlLoaderComplete.xml")).toURI());
         assertNotNull(root, "Import root should not be null");
     }
 

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java
@@ -385,7 +385,14 @@ public class IndentationCheckTest extends AbstractModuleTestSupport {
         checkConfig.addAttribute("lineWrappingIndentation", "4");
         checkConfig.addAttribute("tabWidth", "4");
         checkConfig.addAttribute("throwsIndent", "4");
-        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+        final String[] expected = {
+            "33:9: " + getCheckMessage(MSG_ERROR_MULTI, "operator new", 8, "10, 12"),
+            "40:9: " + getCheckMessage(MSG_ERROR_MULTI, "operator new", 8, "10, 12"),
+            "90:11: " + getCheckMessage(MSG_ERROR_MULTI, "operator new", 10, "12, 14"),
+            "97:13: " + getCheckMessage(MSG_ERROR_MULTI, "operator new", 12, "14, 16"),
+            "119:13: " + getCheckMessage(MSG_ERROR_MULTI, "operator new", 12, "14, 16"),
+            "126:15: " + getCheckMessage(MSG_ERROR_MULTI, "operator new", 14, "16, 18"),
+        };
         verifyWarns(checkConfig, getPath("InputIndentationCorrectIfAndParameter.java"), expected);
     }
 
@@ -658,7 +665,9 @@ public class IndentationCheckTest extends AbstractModuleTestSupport {
             "100:11: " + getCheckMessage(MSG_CHILD_ERROR, "if", 10, 12),
             "101:7: " + getCheckMessage(MSG_ERROR, "if rcurly", 6, 8),
             "104:11: " + getCheckMessage(MSG_ERROR, "Arrays", 10, 12),
+            "110:15: " + getCheckMessage(MSG_ERROR, "operator new", 14, 16),
             "113:11: " + getCheckMessage(MSG_CHILD_ERROR, "method call", 10, 12),
+            "118:15: " + getCheckMessage(MSG_ERROR, "operator new", 14, 16),
             "122:11: " + getCheckMessage(MSG_ERROR, "new", 10, 12),
             "126:11: " + getCheckMessage(MSG_ERROR, "new", 10, 12),
             "127:7: " + getCheckMessage(MSG_ERROR, ")", 6, 8),
@@ -860,6 +869,32 @@ public class IndentationCheckTest extends AbstractModuleTestSupport {
         final String fileName = getPath("InputIndentationValidSwitchIndent.java");
         final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
         verifyWarns(checkConfig, fileName, expected);
+    }
+
+    @Test
+    public void testNewKeyword() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(IndentationCheck.class);
+
+        checkConfig.addAttribute("basicOffset", "4");
+        checkConfig.addAttribute("forceStrictCondition", "false");
+        checkConfig.addAttribute("lineWrappingIndentation", "8");
+        checkConfig.addAttribute("tabWidth", "4");
+        checkConfig.addAttribute("throwsIndent", "8");
+        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+        verifyWarns(checkConfig, getPath("InputIndentationNew.java"), expected);
+    }
+
+    @Test
+    public void testNewChildren() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(IndentationCheck.class);
+
+        checkConfig.addAttribute("basicOffset", "4");
+        checkConfig.addAttribute("forceStrictCondition", "false");
+        checkConfig.addAttribute("lineWrappingIndentation", "8");
+        checkConfig.addAttribute("tabWidth", "4");
+        checkConfig.addAttribute("throwsIndent", "8");
+        final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
+        verifyWarns(checkConfig, getPath("InputIndentationValidNewChildren.java"), expected);
     }
 
     @Test
@@ -1802,13 +1837,13 @@ public class IndentationCheckTest extends AbstractModuleTestSupport {
         checkConfig.addAttribute("throwsIndent", "4");
         checkConfig.addAttribute("arrayInitIndent", "4");
         final String[] expected = {
-            "40:19: " + getCheckMessage(MSG_ERROR_MULTI, "object def rcurly", 18, "8, 12, 16"),
+            "40:19: " + getCheckMessage(MSG_ERROR_MULTI, "object def rcurly", 18, "16, 20, 24"),
             "42:15: " + getCheckMessage(MSG_ERROR, "new", 14, 16),
-            "48:15: " + getCheckMessage(MSG_ERROR_MULTI, "object def rcurly", 14, "8, 12, 16"),
-            "60:19: " + getCheckMessage(MSG_ERROR_MULTI, "object def lcurly", 18, "8, 12, 16"),
-            "66:19: " + getCheckMessage(MSG_ERROR_MULTI, "object def rcurly", 18, "8, 12, 16"),
-            "69:15: " + getCheckMessage(MSG_ERROR_MULTI, "object def lcurly", 14, "8, 12, 16"),
-            "75:15: " + getCheckMessage(MSG_ERROR_MULTI, "object def rcurly", 14, "8, 12, 16"),
+            "48:15: " + getCheckMessage(MSG_ERROR_MULTI, "object def rcurly", 14, "16, 20, 24"),
+            "60:19: " + getCheckMessage(MSG_ERROR_MULTI, "object def lcurly", 18, "16, 20, 24"),
+            "66:19: " + getCheckMessage(MSG_ERROR_MULTI, "object def rcurly", 18, "16, 20, 24"),
+            "69:15: " + getCheckMessage(MSG_ERROR_MULTI, "object def lcurly", 14, "16, 20, 24"),
+            "75:15: " + getCheckMessage(MSG_ERROR_MULTI, "object def rcurly", 14, "16, 20, 24"),
         };
         verifyWarns(checkConfig,
             getPath("InputIndentationAnonymousClassInMethodCurlyOnNewLine.java"), expected);

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationAnonymousClassInMethodCurlyOnNewLine.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationAnonymousClassInMethodCurlyOnNewLine.java
@@ -37,7 +37,7 @@ public class InputIndentationAnonymousClassInMethodCurlyOnNewLine //indent:0 exp
                     { //indent:20 exp:20
                         return null; //indent:24 exp:24
                     } //indent:20 exp:20
-                  }; //indent:18 exp:8,12,16 warn
+                  }; //indent:18 exp:16,20,24 warn
         final Supplier<Map<String, List<AbstractExpressionHandler>>> sup3 = //indent:8 exp:8
               new Supplier<Map<String,List<AbstractExpressionHandler>>>(){ //indent:14 exp:>=16 warn
                     @Override //indent:20 exp:20
@@ -45,7 +45,7 @@ public class InputIndentationAnonymousClassInMethodCurlyOnNewLine //indent:0 exp
                     { //indent:20 exp:20
                         return null; //indent:24 exp:24
                     } //indent:20 exp:20
-              }; //indent:14 exp:8,12,16 warn
+              }; //indent:14 exp:16,20,24 warn
         final Supplier<Map<String, List<AbstractExpressionHandler>>> sup4 = //indent:8 exp:8
                 new Supplier<Map<String, List<AbstractExpressionHandler>>>() //indent:16 exp:>=16
                 { //indent:16 exp:16
@@ -57,22 +57,22 @@ public class InputIndentationAnonymousClassInMethodCurlyOnNewLine //indent:0 exp
                 }; //indent:16 exp:16
         final Supplier<Map<String, List<AbstractExpressionHandler>>> sup5 = //indent:8 exp:8
                 new Supplier<Map<String, List<AbstractExpressionHandler>>>() //indent:16 exp:>=16
-                  { //indent:18 exp:8,12,16 warn
+                  { //indent:18 exp:16,20,24 warn
                       @Override //indent:22 exp:22
                       public Map<String, List<AbstractExpressionHandler>> get() //indent:22 exp:22
                       { //indent:22 exp:22
                           return null; //indent:26 exp:26
                       } //indent:22 exp:22
-                  }; //indent:18 exp:8,12,16 warn
+                  }; //indent:18 exp:16,20,24 warn
         final Supplier<Map<String, List<AbstractExpressionHandler>>> sup6 = //indent:8 exp:8
                 new Supplier<Map<String, List<AbstractExpressionHandler>>>() //indent:16 exp:>=16
-              { //indent:14 exp:8,12,16 warn
+              { //indent:14 exp:16,20,24 warn
                   @Override //indent:18 exp:18
                   public Map<String, List<AbstractExpressionHandler>> get() //indent:18 exp:18
                   { //indent:18 exp:18
                       return null; //indent:22 exp:22
                   } //indent:18 exp:18
-              }; //indent:14 exp:8,12,16 warn
+              }; //indent:14 exp:16,20,24 warn
     } //indent:4 exp:4
 } //indent:0 exp:0
 

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationCorrectIfAndParameter.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationCorrectIfAndParameter.java
@@ -30,14 +30,15 @@ class InputIndentationCorrectIfAndParameter { //indent:0 exp:0
     if (conditionSecond(10000000000.0, new //indent:4 exp:4
         SecondClassLongNam7("Looooooooooooo" //indent:8 exp:8
         + "oooooooooooong").getString(new InputIndentationCorrectIfAndParameter(), //indent:8 exp:8
-        new SecondClassLongNam7("loooooooooong"). //indent:8 exp:8
+        new SecondClassLongNam7("loooooooooong"). //indent:8 exp:10,12 warn
         getInteger(new InputIndentationCorrectIfAndParameter(), "long")), "loong") //indent:8 exp:8
         || conditionThird(2048) || conditionFourth(new //indent:8 exp:8
         SecondClassLongNam7("Looooooooooooooo" //indent:8 exp:8
         + "oo").gB(new InputIndentationCorrectIfAndParameter(), false)) || //indent:8 exp:8
         conditionFifth(true, new SecondClassLongNam7(getString(2048, "Looo" //indent:8 exp:8
         + "ooooooooooooooooooooooooooooooooooooooooooong")).gB( //indent:8 exp:8
-        new InputIndentationCorrectIfAndParameter(), true)) || co(false, new //indent:8 exp:8
+        new InputIndentationCorrectIfAndParameter(), true)) //indent:8 exp:10,12 warn
+        || co(false, new //indent:8 exp:8
         SecondClassLongNam7(getString(100000, "Loooooong" //indent:8 exp:8
         + "Fooooooo><"))) || conditionNoArg() //indent:8 exp:8
         || conditionNoArg() || //indent:8 exp:8
@@ -86,14 +87,15 @@ class InputIndentationCorrectIfAndParameter { //indent:0 exp:0
       if (conditionSecond(10000000000.0, new //indent:6 exp:6
           SecondClassLongNam7("Looooooooooooo" //indent:10 exp:10
           + "ooooooong").getString(new InputIndentationCorrectIfAndParameter(), //indent:10 exp:10
-          new SecondClassLongNam7("loooooooooong"). //indent:10 exp:10
+          new SecondClassLongNam7("loooooooooong"). //indent:10 exp:12,14 warn
           getInteger(new InputIndentationCorrectIfAndParameter(), "long")), "l") //indent:10 exp:10
           || conditionThird(2048) || conditionFourth(new //indent:10 exp:10
           SecondClassLongNam7("Looooooooooooooo" //indent:10 exp:10
             + "og").gB(new InputIndentationCorrectIfAndParameter(), false)) || //indent:12 exp:>=10
             conditionFifth(true, new SecondClassLongNam7(getString(2048, "Loo" //indent:12 exp:>=10
             + "ooooooooooooooooooooooooooooooooooooooooooong")).gB( //indent:12 exp:>=10
-            new InputIndentationCorrectIfAndParameter(), true)) ||co(false,new //indent:12 exp:>=10
+            new InputIndentationCorrectIfAndParameter(), true)) //indent:12 exp:14,16 warn
+              ||co(false,new //indent:14 exp:14
               SecondClassLongNam7(getString(100000, "Loooooong" //indent:14 exp:>=10
               + "Fooooooo><"))) || conditionNoArg() //indent:14 exp:>=10
               || conditionNoArg() || //indent:14 exp:>=10
@@ -114,14 +116,15 @@ class InputIndentationCorrectIfAndParameter { //indent:0 exp:0
         if (conditionSecond(10000000000.0, new //indent:8 exp:8
             SecondClassLongNam7("Looooooooooooo" //indent:12 exp:12
             + "oooooong").getString(new InputIndentationCorrectIfAndParameter(), //indent:12 exp:12
-            new SecondClassLongNam7("loooooooooong"). //indent:12 exp:12
+            new SecondClassLongNam7("loooooooooong"). //indent:12 exp:14,16 warn
             getInteger(new InputIndentationCorrectIfAndParameter(), "lg")), "l") //indent:12 exp:12
             || conditionThird(2048) || conditionFourth(new //indent:12 exp:12
             SecondClassLongNam7("Looooooooooooooo" //indent:12 exp:12
               + "o").gB(new InputIndentationCorrectIfAndParameter(), false))|| //indent:14 exp:>=12
               conditionFifth(true, new SecondClassLongNam7(getString(2048, "L" //indent:14 exp:>=12
               + "ooooooooooooooooooooooooooooooooooooooooooong")).gB( //indent:14 exp:>=12
-              new InputIndentationCorrectIfAndParameter(),true))||co(false,new //indent:14 exp:>=12
+              new InputIndentationCorrectIfAndParameter(),true))//indent:14 exp:16,18 warn
+                || co(false, new //indent:16 exp:16
                 SecondClassLongNam7(getString(100000, "Loooooong" //indent:16 exp:>=12
                 + "Fooooooo><"))) || conditionNoArg() //indent:16 exp:>=12
                 || conditionNoArg() || //indent:16 exp:>=12

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationInvalidMethodIndent.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationInvalidMethodIndent.java
@@ -107,7 +107,7 @@ public class InputIndentationInvalidMethodIndent { //indent:0 exp:0
 
         System.identityHashCode("methods are: " + //indent:8 exp:8
             Arrays.asList( //indent:12 exp:>=12
-              new String[] {"method"}).toString()); //indent:14 exp:>=14
+              new String[] {"method"}).toString()); //indent:14 exp:>=16 warn
 
         System.identityHashCode("methods are: " //indent:8 exp:8
           + Arrays.asList( //indent:10 exp:12 warn
@@ -115,7 +115,7 @@ public class InputIndentationInvalidMethodIndent { //indent:0 exp:0
 
         System.identityHashCode("methods are: " //indent:8 exp:8
             + Arrays.asList( //indent:12 exp:>=12
-              new String[] {"method"}).toString()); //indent:14 exp:>=12
+              new String[] {"method"}).toString()); //indent:14 exp:>=16 warn
 
 
         String blah = (String) System.getProperty( //indent:8 exp:8

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationNew.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationNew.java
@@ -1,0 +1,29 @@
+/* Config:                                                                    //indent:0 exp:0
+ * This test-input is intended to be checked using following configuration:   //indent:1 exp:1
+ *                                                                            //indent:1 exp:1
+ * arrayInitIndent = 4                                                        //indent:1 exp:1
+ * basicOffset = 4                                                            //indent:1 exp:1
+ * braceAdjustment = 0                                                        //indent:1 exp:1
+ * caseIndent = 4                                                             //indent:1 exp:1
+ * forceStrictCondition = false                                               //indent:1 exp:1
+ * lineWrappingIndentation = 8                                                //indent:1 exp:1
+ * tabWidth = 4                                                               //indent:1 exp:1
+ * throwsIndent = 4                                                           //indent:1 exp:1
+ *                                                                            //indent:1 exp:1
+ */                                                                           //indent:1 exp:1
+
+package com.puppycrawl.tools.checkstyle.checks.indentation.indentation; //indent:0 exp:0
+
+import java.io.BufferedReader; //indent:0 exp:0
+import java.io.IOException; //indent:0 exp:0
+import java.io.InputStreamReader; //indent:0 exp:0
+
+public class InputIndentationNew { //indent:0 exp:0
+    void test() throws IOException  { //indent:4 exp:4
+        BufferedReader bf =  //indent:8 exp:8
+                new BufferedReader(  //indent:16 exp:16
+                        new InputStreamReader(System.in) {  //indent:24 exp:24
+                            int a = 0; //indent:28 exp:28
+                        }); //indent:24 exp:24
+    } //indent:4 exp:4
+}  //indent:0 exp:0

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationValidNewChildren.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationValidNewChildren.java
@@ -1,0 +1,39 @@
+/* Config:                                                                    //indent:0 exp:0
+ * This test-input is intended to be checked using following configuration:   //indent:1 exp:1
+ *                                                                            //indent:1 exp:1
+ * arrayInitIndent = 4                                                        //indent:1 exp:1
+ * basicOffset = 4                                                            //indent:1 exp:1
+ * braceAdjustment = 0                                                        //indent:1 exp:1
+ * caseIndent = 4                                                             //indent:1 exp:1
+ * forceStrictCondition = false                                               //indent:1 exp:1
+ * lineWrappingIndentation = 8                                                //indent:1 exp:1
+ * tabWidth = 4                                                               //indent:1 exp:1
+ * throwsIndent = 8                                                           //indent:1 exp:1
+ *                                                                            //indent:1 exp:1
+ */                                                                           //indent:1 exp:1
+
+package com.puppycrawl.tools.checkstyle.checks.indentation.indentation; //indent:0 exp:0
+
+public class InputIndentationValidNewChildren { //indent:0 exp:0
+
+    private final StringBuffer mFilter = //indent:4 exp:4
+            new StringBuffer( //indent:12 exp:12
+                    new CharSequence() { //indent:20 exp:20
+                        @Override //indent:24 exp:24
+                        public char charAt(int index) { //indent:24 exp:24
+                            return 'A'; //indent:28 exp:28
+                        } //indent:24 exp:24
+
+                        public int length() { //indent:24 exp:24
+                            return 1; //indent:28 exp:28
+                        } //indent:24 exp:24
+
+                        public CharSequence subSequence(int start, int end) { //indent:24 exp:24
+                            return this; //indent:28 exp:28
+                        } //indent:24 exp:24
+
+                        public String toString() { //indent:24 exp:24
+                            return "Foo"; //indent:28 exp:28
+                        } //indent:24 exp:24
+                    }); //indent:20 exp:20
+} //indent:0 exp:0

